### PR TITLE
Add dhertz keychron map v0

### PR DESCRIPTION
**Pull request by Dhertz**
Add Dhertz keymap for keychron q2 iso rotary

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [ ] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/keychron/q2/rev_0113/config.h [hunk 1]
```diff
@@ -30,4 +30,4 @@
 #define ENCODERS_PAD_B { B5 }
 
 /* Specifies the number of pulses the encoder registers between each detent */
-#define ENCODER_RESOLUTION 2
+#define ENCODER_RESOLUTION 4
```
Comment (Dhertz) [inline keyboards/keychron/q2/rev_0113/config.h:33]: I am not sure if this is just my board, but it was double stepping

Comment (Dhertz) [reply keyboards/keychron/q2/rev_0113/config.h:33]: seems to be consistent here: https://github.com/qmk/qmk_firmware/blob/master/keyboards/keychron/q2/rev_0111/config.h#L33

File: keyboards/keychron/q2/rev_0113/keymaps/dhertz/keymap.c [hunk 2]
```diff
@@ -0,0 +1,93 @@
+#include QMK_KEYBOARD_H
```
Comment (drashna) [inline keyboards/keychron/q2/rev_0113/keymaps/dhertz/keymap.c:1]: Could you add a GPL2+ compatible license header here? 

For instance:
```c
// Copyright 2021 Your Name (@yourgithub)
// SPDX-License-Identifier: GPL-2.0-or-later
```

Or 

https://github.com/qmk/qmk_firmware/blob/6b299fab6e6a3143c061b9ea005253361fcd524b/data/templates/base/keyboard.c#L1-L15

File: keyboards/keychron/q2/rev_0113/config.h (hunk 1 above)
Comment (drashna) [reply keyboards/keychron/q2/rev_0113/config.h:33]: You mean that on 2, it was registering twice?

File: keyboards/keychron/q2/rev_0113/keymaps/dhertz/keymap.c [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+        if ((get_mods() & MOD_BIT(KC_LCMD)) != MOD_BIT(KC_LCMD)) {
+            register_code(KC_LCMD);
+            defer_exec(20, cancel_cmd, NULL);
+        }
+        if (clockwise) {
+            tap_code(KC_TAB);
+        } else {
+            register_code(KC_LSFT);
+            tap_code(KC_TAB);
+            unregister_code(KC_LSFT);
```
Comment (drashna) [inline keyboards/keychron/q2/rev_0113/keymaps/dhertz/keymap.c:77]:
```suggestion
            tap_code16(S(KC_TAB));
```

**Review by drashna — CHANGES_REQUESTED** [review]

File: keyboards/keychron/q2/rev_0113/config.h (hunk 1 above)
Comment (Dhertz) [reply keyboards/keychron/q2/rev_0113/config.h:33]: correct

Comment (drashna) [reply keyboards/keychron/q2/rev_0113/config.h:33]: Ideally, this should be changed at the keymap level, rather than at keyboard level.

Comment (Dhertz) [reply keyboards/keychron/q2/rev_0113/config.h:33]: I think it is just incorrect right now - as it is 4 on the other model - https://github.com/qmk/qmk_firmware/blob/master/keyboards/keychron/q2/rev_0111/config.h#L33

Comment (Dhertz) [reply keyboards/keychron/q2/rev_0113/config.h:33]: Is there a way to change it on the keymap? I think I get an error about overriding a `#define`

Comment (drashna) [reply keyboards/keychron/q2/rev_0113/config.h:33]: `#undef ENCODER_RESOLUTION`

Comment (Dhertz) [reply keyboards/keychron/q2/rev_0113/config.h:33]: how can I convince you it is a misconfiguration in the board, rather than set it for just my keymap?

Comment (drashna) [reply keyboards/keychron/q2/rev_0113/config.h:33]: did the encoder ship with the board?  or as an add-on?

Comment (Dhertz) [reply keyboards/keychron/q2/rev_0113/config.h:33]: Yeah, it ships as a SKU

**Review by drashna — APPROVED** [review]

**Review by fauxpark — APPROVED** [review]

**Pull request merged**